test/common/filesystem/filesystem_impl_test.cc

```diff
@@ -108,17 +108,16 @@ TEST_F(FileSystemImplTest, FileReadToEndDoesNotExist) {
                EnvoyException);
 }
 
-#ifndef WIN32
-// In Windows this method of removing the permissions does not make read fail.
-TEST_F(FileSystemImplTest, FileReadToEndNotReadable) {
-  const std::string data = "test string\ntest";
-  const std::string file_path = TestEnvironment::writeStringToFileForTest("test_envoy", data);
-
-  std::filesystem::permissions(file_path, std::filesystem::perms::owner_all,
-                               std::filesystem::perm_options::remove);
-  EXPECT_THROW(file_system_.fileReadToEnd(file_path), EnvoyException);
-}
-#endif
+// Issue https://github.com/envoyproxy/envoy/issues/25614
+// TEST_F(FileSystemImplTest, FileReadToEndNotReadable) {
+//  const std::string data = "test string\ntest";
+//  const std::string file_path = TestEnvironment::writeStringToFileForTest("test_envoy", data);
+
+//  std::filesystem::permissions(file_path, std::filesystem::perms::owner_all,
+//                               std::filesystem::perm_options::remove);
+//  EXPECT_THROW(file_system_.fileReadToEnd(TestEnvironment::temporaryPath("envoy_this_not_exist")),
```

Comment:
Please make this line `// EXPECT_THROW(file_system_.fileReadToEnd(file_path), EnvoyException);` - otherwise the commented out test has negative value as documentation, because it's actively misleading.

Reply:
Done. PTAL. cc @RyanTheOptimist